src/layman/user/util.py

```diff
@@ -100,3 +102,11 @@ def get_username_suggestions_from_claims(claims):
 def delete_user(username=None):
     authn_redis.delete_user(username)
     delete_whole_user(username)
+
+
+def is_allowed_to_delete_user(username=None):
+    if username is None:
+        return False
+    allowed_roles_and_users = os.getenv('GRANT_DELETE_OTHER_USER', '').split(',')
+    user_roles = role_service.get_user_roles(username=username)
+    return username in allowed_roles_and_users or any(role in allowed_roles_and_users for role in user_roles)
```

Comment:
This functionality is already implemented by `is_user_in_access_rule` from `layman.authz` module, please use it.